### PR TITLE
Fix metrics/reports for cloned campaigns

### PR DESCRIPTION
**Pull request by solocommand**
[JIRA CS-3041](https://southcomm.atlassian.net/browse/CS-3041)

- Resolves reporting inaccuracies by ensuring the campaign ID is requested when querying for creative metrics or reports.
- Ensures that creative IDs will not collide when cloning a campaign

### CODE REVIEW

File: src/services/campaign-creatives.js [hunk 1]
```diff
@@ -86,6 +86,7 @@ module.exports = {
     const campaign = await Campaign.strictFindActiveById(campaignId);
     const creative = campaign.creatives.id(creativeId);
     if (!creative || creative.deleted) throw new Error('No creative was found for the provided ID.');
+    creative.cid = campaign._id;
```
Comment (zarathustra323) [inline src/services/campaign-creatives.js]: `creative.campaignId`

File: src/graph/resolvers/campaign.js [hunk 2]
```diff
@@ -88,7 +88,7 @@ module.exports = {
 
   CampaignCreative: {
     image: creative => Image.findById(creative.imageId),
-    metrics: creative => analytics.retrieveMetrics({ cre: creative._id }),
+    metrics: ({ _id: cre, cid }) => analytics.retrieveMetrics({ cre, cid }),
```
Comment (zarathustra323) [inline src/graph/resolvers/campaign.js]:
```js
metrics: creative => analytics
  .retrieveMetrics({ cre: creative._id, cid: creative.campaignId }),
```

File: src/schema/campaign/index.js [hunk 3]
```diff
@@ -134,6 +134,7 @@ schema.method('clone', async function clone(user) {
     name: `${this.name} copy`,
   };
   ['id', '_id', 'pushId', 'createdAt', 'updatedAt', 'updatedBy', 'createdBy'].forEach(k => delete input[k]);
+  input.creatives.forEach(cre => cre._id = undefined); // eslint-disable-line
```
Comment (zarathustra323) [inline src/schema/campaign/index.js]: Use the more specific eslint disable rule

**Comment by solocommand** [comment]
Screen Shot 2019-09-18 at 11 51 24 AM

confirmed against production data :)

**Pull request merged**